### PR TITLE
End execution of 'getNewMessages' method if the api call is canceled

### PR DESCRIPTION
**Pull request by ma12-co**
Signed-off-by: Marco Ambrosini <marcoambrosini@pm.me>

### CODE REVIEW

File: src/components/MessagesList/MessagesList.vue [hunk 1]
```diff
@@ -205,6 +205,10 @@ export default {
 			const lastKnownMessageId = this.getLastKnownMessageId()
 			const messages = await lookForNewMessages(this.token, lastKnownMessageId)
 			if (messages !== undefined) {
+				// Ends the execution of the method if the api call has been canceled.
+				if (messages === 'canceled') {
```
Comment (skjnldsv) [inline src/components/MessagesList/MessagesList.vue]: shouldn't you try-catch instead?
Cancelling throw an error

Comment (ma12-co) [reply src/components/MessagesList/MessagesList.vue]: It's already try-catch in the services. If it catches the canceled exeption, the call returns the string `canceled`

Comment (skjnldsv) [reply src/components/MessagesList/MessagesList.vue]: That's not the proper way to do this.
A try catch should not be caught anywhere else than where it's requested.
using response to forward an error message like `messages === 'canceled'` is not the way to go :)

Comment (ma12-co) [reply src/components/MessagesList/MessagesList.vue]: so you suggest try catch both here and in the services?

Comment (skjnldsv) [reply src/components/MessagesList/MessagesList.vue]: no, the service should throw, no error handling in services :)

**Pull request merged**